source/server-selection/tests/logging/server-selection-logging.yml

```diff
@@ -0,0 +1,218 @@
+description: "server-selection-logging"
+
+schemaVersion: "1.13"
+
+createEntities:
+  - client:
+      id: &client client
+      uriOptions:
+        retryWrites: false
+        heartbeatFrequencyMS: 500
+        serverSelectionTimeoutMS: 2000 # avoid hanging for 30s in test expecting SS timeout
+      observeLogMessages:
+        serverSelection: info
+      observeEvents:
+        - serverDescriptionChangedEvent
+      useMultipleMongoses: false
+  - database:
+      id: &database database
+      client: *client
+      databaseName: &databaseName logging-tests
+  - collection:
+      id: &collection collection
+      database: *database
+      collectionName: &collectionName server-selection
+  - client:
+      id: &failPointClient failPointClient
+      useMultipleMongoses: false
+  - collection:
+      id: &unsatisfiableRPColl unsatisfiableRPColl
+      database: *database
+      collectionName: &unsatisfiableRPColl unsatisfiableRPColl
+      collectionOptions:
+        readPreference:
+          mode: Secondary
+          tagSets:
+            - { nonexistenttag : a }
+
+tests:
+  - description: "Message is logged when a suitable server is not immediately available"
+    runOnRequirements:
+      - topologies:
+          - single
+          - replicaset
+        minServerVersion: "4.0" # min version for failCommand on mongod
+      - topologies:
+          - sharded
+        minServerVersion: "4.1.5" # min version for failCommand on mongos
+    operations:
+      # sleep to ensure there is sufficient time for all hosts to be initially discovered.
```

Comment:
I don't love this solution, but I was running into issues where depending on timing, the driver would not have a complete view of the topology yet by the time the test started, e.g. it might think the servers were [primary, unknown, unknown] when server selection for the `updateOne` started, which would lead to extraneous log messages because the driver would re-log in the server selection loop each time it changed a secondary's types from unknown -> secondary. 
that said, I'm realizing that I might be able to write this test in topology-specific ways and have each one start with a `waitForEvent` for the server descriptions changing. I'm not sure whether events that could happen prior to the operations starting are guaranteed to be observable though 🤔 it's possible that the driver ends up discovering everything before we get to the operations. I asked about this in #driver-devs.